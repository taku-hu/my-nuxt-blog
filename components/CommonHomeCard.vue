<template>
  <div class="w-full bg-white shadow-md py-10 mx-auto">
    <div class="flex justify-center items-center mb-10">
      <slot class="w-20" name="titleImage" />
      <div class="text-left">
        <h3 class="text-3xl font-bold">{{ cardData.title }}</h3>
        <p class="text-xl font-bold text-gray-600">{{ cardData.subtitle }}</p>
      </div>
    </div>
    <slot />
  </div>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue'

type CardDataType = {
  title: string;
  subtitle: string;
}

export default Vue.extend({
  props: {
    cardData: {
      type: Object as PropType<CardDataType>,
      required: true
    }
  }
})
</script>
