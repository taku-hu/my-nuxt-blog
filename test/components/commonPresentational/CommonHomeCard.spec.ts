import { shallowMount } from '@vue/test-utils'
// @ts-ignore
import CommonHomeCard from '@/components/commonPresentational/CommonHomeCard.vue'

describe('CommonHomeCard.vue', () => {
  const propsData = {
    cardData: {
      title: 'Sample title',
      subtitle: 'Sample subtitle'
    }
  }

  const wrapper = shallowMount(CommonHomeCard, { propsData })
  it('props', () => {
    expect(wrapper.props()).toEqual(propsData)
  })
  it('template', () => {
    expect(wrapper.html()).toContain('<h3 class="text-3xl font-bold">Sample title</h3>')
    expect(wrapper.html()).toContain('<p class="text-xl font-bold text-gray-600">Sample subtitle</p>')
  })
})
