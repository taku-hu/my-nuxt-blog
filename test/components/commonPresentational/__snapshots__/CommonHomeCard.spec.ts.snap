// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonHomeCard.vue snapshot 1`] = `
<div
  class="w-full bg-white shadow-md py-10 mx-auto"
>
  <div
    class="flex justify-center items-center mb-10"
  >
     
    <div
      class="text-left"
    >
      <h3
        class="text-3xl font-bold"
      >
        Sample title
      </h3>
       
      <p
        class="text-xl font-bold text-gray-600"
      >
        Sample sub title
      </p>
    </div>
  </div>
   
</div>
`;
